### ISSUE TITLE
Add support for AWS Sagemaker Endpoint Resources

### ISSUE THREAD

**Issue by ncilfone** [infracost/infracost#2733]
∙ 3 are not supported yet, see https://infracost.io/requested-resources:
  ∙ 1 x aws_sagemaker_endpoint
  ∙ 1 x aws_sagemaker_endpoint_configuration
  ∙ 1 x aws_sagemaker_model

**Comment by nisalupendra** [comment]
Yes, this would be very useful as entire production pipelines running on sagemaker

**Comment by ThibaudMrx** [comment]
It would be extremely usefull, some sagemaker instances can make up to 80% of infra cost

**Comment by muhammadahmed-01** [comment]
**Update: Planning to implement**

I am planning to implement support for `aws_sagemaker_endpoint_configuration`. 

After a thorough review of the [resource schema](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sagemaker_endpoint_configuration), I've identified that this resource requires handling several distinct billing paths. My plan is to cover:

1. **Provisioned Throughput:** Mapping costs based on the `instance_type` and `initial_instance_count` within the `production_variants` block.
2. **Serverless Inference:** Detecting the `serverless_config` block and its required arguments (`max_concurrency`, `memory_size_in_mb`) to estimate usage-based costs.
3. **Multi-Variant & Shadow Deployments:** Ensuring we capture costs for both primary and `shadow_production_variants`.
4. **Async & Data Capture:** Accounting for the additional overhead defined in `async_inference_config` and `data_capture_config`.

I'll be following the standard AWS provider patterns in the Infracost repo. I will link the PR here once the initial mapping logic is ready for review.

**Comment by muhammadahmed-01** [comment]
Raised [PR](https://github.com/infracost/infracost/pull/3502) to implement support for `aws_sagemaker_endpoint_configuration`

**Comment by glenngillen** [comment]
Unfortunately @muhammadahmed-01's contribution was for the now deprecated 0.10.x releases. In part to prevent situations like this (generous people spending their time making contributions we're unable to merge for some reason) we've switched to a model where we're only using GitHub issues for problems that are well understood, defined, prioritised, and that we're aligned on merging in when they land. All feature requests, upvoting on them, etc. is now in GitHub Discussions. 

If this is an area that there is still community interest in supporting please create a Discussion and move further conversations over there.
